src/lang/KclSingleton.ts

```diff
@@ -211,6 +217,24 @@ export class KclManager {
     }
   }
 
+  // (jess) I'm not in love with this, but it ensures we clear the scene and
+  // bust the cache on
+  // errors from parsing when opening new files.
+  // Why not just clear the cache on all parse errors, you ask? well its actually
+  // really nice to keep the cache on parse errors within the same file, and
+  // only bust on engine errors esp if they take a long time to execute and
+  // you hit the wrong key!
+  private async checkIfSwitchedFilesShouldClear() {
+    // If we were switching files and we hit an error on parse we need to bust
+    // the cache and clear the scene.
+    if (this._hasErrors && this._switchedFiles) {
+      void clearSceneAndBustCache(this.engineCommandManager)
```

Comment:
Await this and let the caller decide whether to run this whole thing in the background.

```suggestion
      await clearSceneAndBustCache(this.engineCommandManager)
```

Reply:
okay so if i await this the whole safeParse needs to be awaited and it changes the whole world, are we okay w that

Reply:
okay let me look trhu all the callers